docs: create MCD deployment guide (#100)
makerdao/developerguides: mcd/mcd-deplyoment/mcd-deployment.md

    - [Configuration](#configuration)
      - [Account configuration](#account-configuration)
      - [Chain configuration](#chain-configuration)
    - [MCD Deployment process](#mcd-deployment-process)
      - [Deploy on local testchain with default config file](#deploy-on-local-testchain-with-default-config-file)
      - [Deploy on Goerli with default config file](#deploy-on-goerli-with-default-config-file)
      - [Deploy on Mainnet with default config file](#deploy-on-mainnet-with-default-config-file)
      - [Deploy on any network passing a custom config file](#deploy-on-any-network-passing-a-custom-config-file)
      - [Output](#output)
      - [Helper scripts](#helper-scripts)
  - [Smart Contract Dependencies](#smart-contract-dependencies)
  - [Additional Resources](#additional-resources)

## Overview

This guide will walk you through deploying MCD protocol to your prefered network.

## Learning Objectives

Here you’ll learn how to deploy your own MCD protocol.

- MCD deployment configuration
- Deployment process

## Pre-requisites

Knowledge in:

- [dapp.tools](http://dapp.tools/)
- [makerdao/dss-deploy-scripts](https://github.com/makerdao/dss-deploy-scripts)

> - Testnet ETH kovan/rinkeby/ropsten/goerli

Reviewer: should we just refer to Goerli?

Author: hmm... then we make this guide specific only for Goerli?

Reviewer 2: This can be useful for other networks as well, so I'd keep it generic.

## Sections

> 

Reviewer: Do we need a section detailing RPC Nodes to use ie. Alchemy and or local machine (I can provide my Geth Script) - but we may want to point out the pros and cons of a local goerli instance.

Reviewer 2: In my experience, remote RPC nodes are more likely to simply drop your txs randomly... this will cause the script to fail and unfortunately you need to start from scratch.

Author: I think that the one who will be using this guide already knows how to deal with RPC nodes and which one to use

### Installing

> First if all we will need to clone `dss-deploy-script` repo:

Reviewer: First of all ..

```shell
git clone https://github.com/makerdao/dss-deploy-scripts.git
cd dss-deploy-scripts
```

> The only way to install everything necessary to deploy is Nix. Run

Reviewer: No we need to add a dependency section - which would include dapp tools and NIX and links for installing those ?

Author: Fixed

    - [MCD Deployment process](#mcd-deployment-process)
      - [Deploy on local testchain with default config file](#deploy-on-local-testchain-with-default-config-file)
      - [Deploy on Goerli with default config file](#deploy-on-goerli-with-default-config-file)
      - [Deploy on Mainnet with default config file](#deploy-on-mainnet-with-default-config-file)
      - [Deploy on any network passing a custom config file](#deploy-on-any-network-passing-a-custom-config-file)
      - [Output](#output)
      - [Helper scripts](#helper-scripts)
  - [Smart Contract Dependencies](#smart-contract-dependencies)
  - [Additional Resources](#additional-resources)

## Overview

This guide will walk you through deploying MCD protocol to your prefered network.

## Learning Objectives

Here you’ll learn how to deploy your own MCD protocol.

- MCD deployment configuration
- Deployment process

## Pre-requisites

Knowledge in:

- [dapp.tools](http://dapp.tools/)
- [makerdao/dss-deploy-scripts](https://github.com/makerdao/dss-deploy-scripts)
- Testnet ETH kovan/rinkeby/ropsten/goerli

### Dependencies

We need to have `nix` and `dapp.tools` installed on our machine.

> - Install `nix`: You can use this [instruction](https://nixos.org/download.html)

Reviewer 2: ```suggestion
- Install `nix`: You can use these [instructions](https://nixos.org/download.html)
```

- Install `dapp.tools`: ```curl https://dapp.tools/install | sh```

## Sections

### Installation

First of all we will need to clone `dss-deploy-script` repo:

> ```shell

Reviewer 2: ```suggestion
```bash
```

git clone https://github.com/makerdao/dss-deploy-scripts.git
cd dss-deploy-scripts
```

The only way to install everything necessary to deploy MCD we need to run

> ```shell

Reviewer 2: ```suggestion
```bash
```

  "dog_hole": "<Max total DAI needed to cover all debt plus penalty fees on active Clip auctions in DAI unit>",
  "jug_base": "<Base component of stability fee in percentage per year (e.g. 2.5)>",
  "pot_dsr": "<Dai Savings Rate in percentage per year (e.g. 2.5)>",
  "end_wait": "<Global Settlement cooldown period in seconds>",
  "esm_pit": "<Pit address to send MKR to be burnt when ESM is fired>",
  "esm_min": "<Minimum amount to trigger ESM in MKR unit>",
  "flap_beg": "<Minimum bid increase in percentage (e.g. 5.5)>",
  "flap_ttl": "<Max time between bids in seconds>",
  "flap_tau": "<Max auction duration in seconds>",
  "flop_beg": "<Minimum bid increase in percentage (e.g. 5.5)>",
  "flop_pad": "<Increase of lot size after `tick` in percentage (e.g. 50)>",
  "flop_ttl": "<Max time between bids in seconds>",
  "flop_tau": "<Max auction duration in seconds>",
  "flap_lid": "<Max amount of DAI that can be put up for sale at the same time in DAI unit (e.g. 1000000)>",
  "flash_max": "<Max DAI can be borrowed from flash loan module in DAI unit (e.g. 1000000)>",
  "flash_toll": "<Fee being charged from amount being borrow via flash loan module in percentage (e.g 0.1%)>",

>   import: {

Reviewer 2: ```suggestion
  "import": {
```

            "cusp": "<Percentage taken for the new price before auction reset (e.g. 30%)>",
            "calc": {
              "type": "LinearDecrease/StairstepExponentialDecrease/ExponentialDecrease",
              "tau":  "<Time after auction start when the price reaches zero in seconds (LinearDecrease)>",
              "step": "<Length of time between price drops in seconds (StairstepExponentialDecrease)>",
              "cut":  "<Percentage to be taken as new price per step (e.g. 99%, which is 1% drop) (StairstepExponentialDecrease/ExponentialDecrease)>"
            },
            "cm_tolerance": "<Percentage of previous price which a drop would enable anyone to be able to circuit break the liquidator via ClipperMom (e.g. 50%)>"
          }
        }
      }
    }
  }
}
```

<div style="padding: 15px; border: 1px solid transparent; border-color: transparent; margin-bottom: 20px; border-radius: 4px; color: #8a6d3b;; background-color: #fcf8e3; border-color: #faebcc;">
<h5 style="color: #8a6d3b">Note</h5>
Make sure to launch <bold style="color: grey">MCD_ADMIN</bold> if you are providing it in <bold style="color: grey">config.authority</bold>.
</div>

> ```shell
> sethSend "$MCD_GOV" 'mint(address,uint256)' "$ETH_FROM" "$(seth --to-uint256 "$(seth --to-wei 1000000 ETH)")"
> ```

Reviewer 2: ```suggestion

```bash
sethSend "$MCD_GOV" 'mint(address,uint256)' $ETH_FROM $(seth --to-wei 1000000 ETH)
```
```

<div style="padding: 15px; border: 1px solid transparent; border-color: transparent; margin-bottom: 20px; border-radius: 4px; color: #8a6d3b;; background-color: #fcf8e3; border-color: #faebcc;">
<h5 style="color: #8a6d3b">Note</h5>

> Make sure to launch <bold style="color: grey">MCD_ADMIN</bold> if you are providing it in <bold style="color: grey">config.authority</bold>.

Reviewer 2: ```suggestion
Make sure to launch `MCD_ADMIN` if you are providing it in `config.authority`.
```

Author: This want work at least for me, as this text inside of html

</div>

> ```shell

Reviewer 2: ```suggestion

```bash
```

# lock enough MKR (80,000 MKR threshold)

> sethSend "$MCD_GOV" "approve(address,uint256)" "$MCD_ADM" "$(seth --to-uint256 "$(seth --to-wei 80000 ETH)")"
> sethSend "$MCD_ADM" "lock(uint256)" "$(seth --to-uint256 "$(seth --to-wei 80000 ETH)")"

Reviewer 2: ```suggestion
sethSend "$MCD_GOV" "approve(address,uint256)" "$MCD_ADM" $(seth --to-wei 80000 ETH)
sethSend "$MCD_ADM" "lock(uint256)" $(seth --to-wei 80000 ETH)
```
